pkg/kubectl/service.go

```diff
@@ -134,10 +147,14 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 		if len(portStringSlice) > 1 {
 			name = fmt.Sprintf("port-%d", i+1)
 		}
+		protocol := params["protocol"]
```

Comment:
IIUC, you mean the priority order, as for now, I will always try to get a protocol from the port protocol map, and if has no such a mapping for a port, I will use the --protocol flag value.